### PR TITLE
Fix name of attribute requirement repository + fix requirements management

### PR DESCRIPTION
**Pull request by rybus**
**Description (for Contributor and Core Developer)**

This PR fixes a typo in the attribute requirement repository definition of service.

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | N/A
| Added Behats                      | N/A
| Changelog updated                 | No
| Review and 2 GTM                  | Yes
| Micro Demo to the PO (Story only) | N/A
| Migration script                  | N/A
| Tech Doc                          | N/A

### CODE REVIEW

**Comment by damien-carcel** [comment]
GTM

**Comment by ClementGautier** [comment]
GTM

File: src/Pim/Bundle/CatalogBundle/Resources/config/updaters.yml [hunk 1]
```diff
@@ -59,108 +59,109 @@ parameters:
...
     pim_catalog.updater.family:
-        class: %pim_catalog.updater.family.class%
+        class: '%pim_catalog.updater.family.class%'
         arguments:
             - '@pim_catalog.repository.family'
             - '@pim_catalog.factory.family'
             - '@pim_catalog.repository.attribute'
             - '@pim_catalog.repository.channel'
             - '@pim_catalog.factory.attribute_requirement'
+            - '@pim_catalog.repository.attribute_requirement'
```
Comment (rybus) [inline src/Pim/Bundle/CatalogBundle/Resources/config/updaters.yml]: This is where I actually added something ;)

File: src/Pim/Component/Catalog/Updater/FamilyUpdater.php [hunk 2]
```diff
@@ -4,7 +4,9 @@
 
 use Akeneo\Component\StorageUtils\Repository\IdentifiableObjectRepositoryInterface;
 use Akeneo\Component\StorageUtils\Updater\ObjectUpdaterInterface;
+use Doctrine\Common\Collections\ArrayCollection;
 use Doctrine\Common\Util\ClassUtils;
+use Doctrine\ORM\EntityRepository;
```
Comment (skeleton) [inline src/Pim/Component/Catalog/Updater/FamilyUpdater.php]: You cannot add a reference to ORM in a Pim Component

File: src/Pim/Component/Catalog/Updater/FamilyUpdater.php [hunk 3]
```diff
@@ -43,26 +45,32 @@ class FamilyUpdater implements ObjectUpdaterInterface
...
+     * @param EntityRepository                      $requirementRepo
      */
     public function __construct(
         IdentifiableObjectRepositoryInterface $familyRepository,
         FamilyFactory $familyFactory,
         AttributeRepositoryInterface $attributeRepository,
         ChannelRepositoryInterface $channelRepository,
-        AttributeRequirementFactory $attrRequiFactory
+        AttributeRequirementFactory $attrRequiFactory,
+        EntityRepository $requirementRepo
```
Comment (skeleton) [inline src/Pim/Component/Catalog/Updater/FamilyUpdater.php]: AttributeRequirementRepositoryInterface $requirementRepo

File: src/Pim/Component/Catalog/Updater/FamilyUpdater.php [hunk 4]
```diff
@@ -163,15 +191,17 @@ protected function getExistingIdentifierRequirements(FamilyInterface $family)
     /**
      * Creates attribute requirements for the given channel but skip identifiers
      *
-     * @param array  $attributeCodes
-     * @param string $channelCode
-     *
-     * @throws \InvalidArgumentException
```
Comment (skeleton) [inline src/Pim/Component/Catalog/Updater/FamilyUpdater.php]: why do you remove @throws param ?

File: src/Pim/Component/Catalog/Updater/FamilyUpdater.php [hunk 5]
```diff
@@ -205,30 +235,38 @@ protected function addMissingIdentifierRequirements(array $requirements)
...
 
         return $requirements;
     }
 
     /**
+     * @param FamilyInterface    $family
      * @param AttributeInterface $attribute
      * @param string             $channelCode
      *
-     * @throws \InvalidArgumentException
```
Comment (skeleton) [inline src/Pim/Component/Catalog/Updater/FamilyUpdater.php]: same

File: src/Pim/Component/Catalog/Updater/FamilyUpdater.php [hunk 6]
```diff
@@ -205,30 +235,38 @@ protected function addMissingIdentifierRequirements(array $requirements)
...
         if (null === $channel) {
             throw new \InvalidArgumentException(
                 sprintf('Channel with "%s" code does not exist', $channelCode)
             );
         }
+        $requirement = $this->requirementRepo->findOneBy(
+            [
+                'attribute' => $attribute->getId(),
+                'channel'   => $channel->getId(),
+                'family'    => $family->getId()]
```
Comment (skeleton) [inline src/Pim/Component/Catalog/Updater/FamilyUpdater.php]: `]` on a new line please

File: src/Pim/Component/Catalog/Updater/FamilyUpdater.php [hunk 7]
```diff
@@ -130,20 +138,40 @@ protected function setLabels(FamilyInterface $family, array $data)
...
+        $this->removeRequirements($family, $requirements, $oldRequirements);
         $family->setAttributeRequirements($requirements);
     }
 
     /**
      * @param FamilyInterface $family
+     * @param array           $requirements
+     * @param array           $oldRequirements
+     */
+    protected function removeRequirements(
```
Comment (skeleton) [inline src/Pim/Component/Catalog/Updater/FamilyUpdater.php]: can you explain why do you do this please ?

File: src/Pim/Component/Catalog/Updater/FamilyUpdater.php [hunk 8]
```diff
@@ -163,15 +191,17 @@ protected function getExistingIdentifierRequirements(FamilyInterface $family)
...
-     * @param array  $attributeCodes
-     * @param string $channelCode
-     *
-     * @throws \InvalidArgumentException
+     * @param FamilyInterface $family
+     * @param array           $attributeCodes
+     * @param string          $channelCode
      *
-     * @return AttributeRequirementInterface[]
+     * @return array
```
Comment (skeleton) [inline src/Pim/Component/Catalog/Updater/FamilyUpdater.php]: as you return an array of AttributeRequirementInterface, please put back `@return AttributeRequirementInterface[]`

File: src/Pim/Component/Catalog/Updater/FamilyUpdater.php (hunk 7 above)
Comment (rybus) [inline src/Pim/Component/Catalog/Updater/FamilyUpdater.php]: this is how it is done on 1.5 branch

File: src/Pim/Bundle/CatalogBundle/Repository/AttributeRequirementRepositoryInterface.php [hunk 9]
```diff
@@ -0,0 +1,16 @@
+<?php
```
Comment (skeleton) [inline src/Pim/Bundle/CatalogBundle/Repository/AttributeRequirementRepositoryInterface.php]: you can put this interface in `Pim\Component\Catalog\Repositor` namespace

**Comment by skeleton** [comment]
GTM once fixed. Thanks a lot for this PR :+1:

**Comment by damien-carcel** [comment]
GTM once Skeleton comment fixed.

**Pull request merged**